src/main/java/org/springframework/data/couchbase/core/mapping/CouchbasePersistentEntity.java

```diff
@@ -40,6 +43,26 @@
 	 */
 	int getExpiry();
 
+	/**
+	 * Returns the expiration time of the entity.
+	 * <p/>
+	 * The Couchbase format for expiration time is: - for TTL < 31 days (<= 30 * 24 * 60 * 60): expressed as a TTL in
+	 * seconds - for TTL > 30 days: expressed as Unix UTC time of expiry (number of SECONDS since the Epoch)
+	 *
+	 * @return the expiration time Duration
+	 */
+	Duration getExpiryDuration();
+
+	/**
+	 * Returns the expiration time of the entity.
+	 * <p/>
+	 * The Couchbase format for expiration time is: - for TTL < 31 days (<= 30 * 24 * 60 * 60): expressed as a TTL in
+	 * seconds - for TTL > 30 days: expressed as Unix UTC time of expiry (number of SECONDS since the Epoch)
+	 *
+	 * @return the expiration time Instant
+	 */
+	Instant getExpiryInstant();
```

Comment:
Is this method required? It seems like the value it returns is calculated from the _duration_ specified by the user in the annotation.

If it's only used by integration tests, perhaps it could be moved into the test code?

Reply:
Removed.